README.md

```diff
@@ -1,22 +1,43 @@
 
-## Intro
+![Architecture
+Diagram](http://mesosphere.github.io/mesos-dns/img/architecture.png)
 
-[Mesos-DNS](https://github.com/mesosphere/mesos-dns) supports service discovery in [Apache Mesos](http://mesos.apache.org/) clusters. It allows applications and services running on Mesos to find each other through the domain name system ([DNS](http://en.wikipedia.org/wiki/Domain_Name_System)), similar to how services discover each other throughout the Internet. Applications launched by [Marathon](https://github.com/mesosphere/marathon) or [Aurora](http://aurora.incubator.apache.org/) are assigned names like `search.marathon.mesos` or `log-aggregator.aurora.mesos`. Mesos-DNS translates these names to the IP address and port on the machine currently running each application. To connect to an application in the Mesos datacenter, all you need to know is its name. 
+## Status
+This project is undergoing heavy refactoring, hardening and testing and
+is considered **alpha**. We have adopted a [semantic versioning](http://semver.org/) scheme which clearly communicates its status and progression over time.
 
-## Getting started
+## Installing
+The official distribution and installation channel is pre-compiled binaries available in [Github releases](https://github.com/mesosphere/mesos-dns/releases).
 
-Read the [docs](http://mesosphere.github.io/mesos-dns/docs/) for setup instructions.
+## Building
+Building the **master** branch from source should always succeed but doesn't provide
+the same stability and compatibility guarantees as releases.
 
-## How it Works
+All dependencies are vendored using `Godeps` which require you to
+install it in order to build from source.
 
-Mesos-DNS is designed to be a minimal, stateless service that is easy to deploy and maintain. The figure below depicts how it works:
+```shell
+$ go get github.com/tools/godep
+$ godep go build ./...
+```
 
-![Mesos-DNS architecture](https://github.com/mesosphere/mesos-dns/blob/master/docs/mesos-dns.png)
+## Testing
+```shell
+$ godep go test -race ./...
+```
 
-Mesos-DNS periodically queries the Mesos master(s), retrieves the state of all running tasks from all running frameworks, and generates DNS records for these tasks (A and SRV records). As tasks start, finish, fail, or restart on the Mesos cluster, Mesos-DNS updates the DNS records to reflect the latest state. The configuration of Mesos-DNS is minimal. You simply point it to the Mesos masters at launch. Frameworks do not need to communicate with Mesos-DNS at all. Applications and services running on Mesos slaves can discover the IP addresses and ports of other applications they depend upon by issuing DNS lookup requests. Mesos-DNS replies directly to DNS requests for tasks launched by Mesos. For DNS requests for other hostnames or services, Mesos-DNS uses an external nameserver to derive replies.
```

Comment:
This is probably important for people who are trying to differentiate among a set of options for implementing service discovery. Can we move this somewhere else instead of deleting it?